### PR TITLE
Allow unsetting target technology

### PR DESCRIPTION
**Pull request by exyi**
AFAIK, target technology option could not be set to null from GUI - to all researches.

### CODE REVIEW

File: Yafc/Windows/PreferencesScreen.cs [hunk 1]
```diff
@@ -89,11 +89,11 @@ public class PreferencesScreen : PseudoScreen {
         /// <summary>Add a GUI element that opens a popup to allow the user to choose from the <paramref name="list"/>, which triggers <paramref name="select"/>.</summary>
         /// <param name="text">Label to show.</param>
         /// <param name="width">Width of the popup. Make sure it is wide enough to fit text!</param>
-        private void ChoiceObject<T>(ImGui gui, string text, T[] list, T current, Action<T> select, float width = 20f) where T : FactorioObject {
+        private void ChoiceObject<T>(ImGui gui, string text, T[] list, T current, Action<T> select, float width = 20f, bool allowNone = false) where T : FactorioObject {
```
Comment (veger) [inline Yafc/Windows/PreferencesScreen.cs:92]: We are trying to improve the documentation little by little (no one volunteers to update all in one go :wink: ), so could you please document the new parameter you added to this function as another little step?

**Comment by have-fun-was-taken** [comment]
Hi @exyi! Thank you for the contribution!
Could you please explain what happens with cost analysis if the target is set to null? It would be awesome if you added your understanding as a message to your commit and as documentation to the `CostAnalysis` class.

**Comment by exyi** [comment]
Sure! The setting sets the resource objectives for the linear programming solver. Most importantly, YAFC then shows the estimated required resource amounts to something I'll actually need soon-ish, rather than in 1000 hours :sweat_smile:. But it also affects the costs slightly, in this case it's probably that more aluminium processing byproducts will be needed after logi, making aluminium itself cheaper.

image
image

**Comment by have-fun-was-taken** [comment]
Could you please describe the behavior before and after the change?
On my side, I always get `Estimated amount for all researches` no matter if I select nothing or not.

**Comment by exyi** [comment]
You have to save project and restart YAFC (or also change mining productivity...). That's another glitch that YAFC doesn't update the cost analysis when preferences change, only when settings change (same thing as Dorus encountered with the pollution cost flag https://github.com/Dorus/yafc-ce/pull/1). Now, the option is to also enable change event for preferences, or move this into settings. Do you know the difference between the two?

**Review by have-fun-was-taken — APPROVED** [review]

**Pull request merged**